scipy/stats/qmc.py

```diff
@@ -1104,7 +1113,15 @@ def random(self, n_samples=1):
 
         """
         sample = np.empty((n_samples, self.dim), dtype=float)
-        _draw(n_samples, self.num_generated, self.dim, self._sv, self._quasi, sample)
+
+        if self.num_generated == 0 and n_samples == 1:
+            sample = self._first_point
+        elif self.num_generated == 0:
+            _draw(n_samples, self.num_generated, self.dim, self._sv, self._quasi, sample)
+            sample = np.concatenate([self._first_point, sample])[:n_samples]
```

Comment:
Make logic slightly easier to follow 

```suggestion
        if self.num_generated == 0:
            if n_samples == 1:
                sample = self._first_point
            else:
                _draw(n_samples, self.num_generated, self.dim, self._sv, self._quasi, sample)
                sample = np.concatenate([self._first_point, sample])[:n_samples]
```